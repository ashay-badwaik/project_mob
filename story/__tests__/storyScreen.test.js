import 'react-native';
import React from 'react';
import StoryScreen from './../components/storyScreen';
import AsyncStorage from '@react-native-async-storage/async-storage';


// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

jest.useFakeTimers();


const mockedParams = {
    navigation:{
        navigate: jest.fn()
    }
};

const tree = renderer.create(
    <StoryScreen {...mockedParams}/>
)

it('renders correctly', () => {
    tree
})

it('snapshot', () => {
    expect(tree).toMatchSnapshot()
})

it('components render', () => {
    const img = tree.root.findByProps({testID: 'img'})
    const bar = tree.root.findByProps({testID: 'progress'})
    const con = tree.root.findByProps({testID: 'content'})
    expect(img).not.toBeNull()
    expect(bar).not.toBeNull()
    expect(con).not.toBeNull()
})

