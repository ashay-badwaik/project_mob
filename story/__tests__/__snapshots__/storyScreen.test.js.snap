// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#17588a",
      "height": "100%",
    }
  }
>
  <View
    animated={true}
    animationConfig={
      Object {
        "bounciness": 0,
      }
    }
    animationType="spring"
    indeterminate={false}
    indeterminateAnimationDuration={1000}
    onLayout={[Function]}
    progress={0}
    style={
      Array [
        Object {
          "backgroundColor": undefined,
          "borderColor": "#ffffff",
          "borderRadius": 4,
          "borderWidth": 1,
          "overflow": "hidden",
          "width": 300,
        },
        Object {
          "alignSelf": "center",
          "marginHorizontal": 30,
          "marginVertical": 20,
          "width": 300,
        },
      ]
    }
    testID="progress"
    useNativeDriver={false}
  >
    <View
      style={
        Object {
          "backgroundColor": "#ffffff",
          "height": 6,
          "transform": Array [
            Object {
              "translateX": 0,
            },
            Object {
              "translateX": -149,
            },
            Object {
              "scaleX": 0.0001,
            },
          ],
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderRadius": 30,
        "height": 300,
        "justifyContent": "center",
        "marginHorizontal": 30,
        "marginVertical": 40,
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "",
        }
      }
      style={
        Object {
          "borderRadius": 30,
          "height": "100%",
          "width": "100%",
        }
      }
      testID="img"
    />
  </View>
  <Text
    style={
      Object {
        "color": "#fff",
        "fontSize": 30,
        "margin": 30,
        "textAlign": "center",
      }
    }
    testID="content"
  >
    
  </Text>
</View>
`;
